python_modules/dagster/dagster_tests/core_tests/test_pipeline_execution.py

```diff
@@ -255,7 +256,9 @@ def test_singleton_pipeline():
     def single_solid_pipeline():
         stub_solid()
 
-    assert execute_pipeline(single_solid_pipeline).success
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        assert execute_pipeline(single_solid_pipeline).success
```

Comment:
does this fail before the change? not obvious from api names. Worth a comment as well.

Reply:
Yes it does.

Unfortunately `pytest` does not offer a better API for ensuring that no warnings are emitted: https://docs.pytest.org/en/latest/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests

Reply:
kk - worth leaving a comment 

* would be good to do this using modern APIs (job/op/graph). `execute_in_process` should trigger the issue just as ell as execute_pipeline does
* I would include the defining of the job/op/graph objects in the no-warnings context manager